seismicpro/src/gather/gather.py

```diff
@@ -366,10 +363,12 @@ def dump(self, path, name=None, retain_parent_segy_headers=True):
 
             # Dump traces and their headers. Optionally copy headers from the parent SEG-Y file.
             dump_handler.trace = self.data
-            for i, dump_h in trace_headers_dict.items():
+            for i, trace_headers in enumerate(self[used_header_names]):
                 if retain_parent_segy_headers:
-                    dump_handler.header[i].update(parent_handler.header[trace_ids[i]])
-                dump_handler.header[i].update({**dump_h, segyio.TraceField.TRACE_SAMPLE_INTERVAL: sample_rate})
+                    dump_handler.header[i] = parent_handler.header[trace_ids[i]]
+                dump_handler.header[i].update({**dict(zip(used_header_bytes, trace_headers)),
+                                               segyio.TraceField.TRACE_SAMPLE_INTERVAL: sample_rate,
+                                               segyio.TraceField.TRACE_SEQUENCE_FILE: i})
```

Comment:
```suggestion
                                               segyio.TraceField.TRACE_SEQUENCE_FILE: i+1})
```
TSF starts with 1, strange that it worked